kolibri/plugins/coach/assets/src/composables/useQuizCreation.js

```diff
@@ -156,9 +156,32 @@ export default function useQuizCreation(DEBUG = false) {
     if (!targetSection) {
       throw new TypeError(`Section with id ${section_id} not found; cannot be updated.`);
     }
-    const { question_count } = updates;
-    if (question_count) {
-      if (question_count < (targetSection.question_count || 0)) {
+    // Make sure all questions in the an updated pool have proper unique IDs
```

Comment:
Possible additional refactoring in the future - the resource pool and questions could maybe be factored out into separate functions, that are both called from this one?